File: launchschool_exercises/210_small_problems/list_processing/grocery-list.js
/*
  Write a function that takes a grocery list (a two-dimensional array) with each element containing a fruit and a quantity, and returns a one-dimensional array of fruits, in which each fruit appears a number of times equal to its quantity.
*/

function buyFruit(fruitList) {
  var fruits = [];

  fruitList.forEach(function (fruit) {
    for (let i = 0; i < fruit[1]; i++) {
      fruits.push(fruit[0]);
    }
  });

  return fruits;
}

console.log(buyFruit([['apple', 3], ['orange', 1], ['banana', 2]]));
// returns ["apple", "apple", "apple", "orange", "banana", "banana"]

File: launchschool_exercises/109_small_problems/easy9/grocery_list.rb
def buy_fruit(list)
  new_list = []
  list.each do |fruit|
    fruit[1].times { new_list << fruit[0] }
  end
  new_list
end

p buy_fruit([["apples", 3], ["orange", 1], ["bananas", 2]]) ==
  ["apples", "apples", "apples", "orange", "bananas","bananas"]
